src/Avalonia.Controls/SplitButton/ToggleSplitButton.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using Avalonia.Controls.Primitives;
+using Avalonia.Interactivity;
+using Avalonia.Styling;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// A button with primary and secondary parts that can each be pressed separately.
+    /// The primary part behaves like a <see cref="ToggleButton"/> with two states and
+    /// the secondary part opens a flyout.
+    /// </summary>
+    public class ToggleSplitButton : SplitButton, IStyleable
+    {
+        /// <summary>
+        /// Raised when the <see cref="IsChecked"/> property value changes.
+        /// </summary>
+        public event EventHandler<RoutedEventArgs> IsCheckedChanged
+        {
+            add => AddHandler(IsCheckedChangedEvent, value);
+            remove => RemoveHandler(IsCheckedChangedEvent, value);
+        }
+
+        /// <summary>
+        /// Defines the <see cref="IsCheckedChanged"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> IsCheckedChangedEvent =
+            RoutedEvent.Register<ToggleSplitButton, RoutedEventArgs>(
+                nameof(IsCheckedChanged),
+                RoutingStrategies.Bubble);
+
+        /// <summary>
+        /// Defines the <see cref="IsChecked"/> property.
+        /// </summary>
+        public static readonly StyledProperty<bool> IsCheckedProperty =
+            AvaloniaProperty.Register<ToggleSplitButton, bool>(
+                nameof(IsChecked));
```

Comment:
I could not AddOwner here using the ToggleButton.IsCheckedProperty. The reason is ToggleButton is three state; however, ToggleSplitButton is only two state. I think this is fine for now.

Reply:
Comment was made here: https://github.com/AvaloniaUI/Avalonia/pull/7422#issuecomment-1021911616

> ToggleSplitButton.IsCheckedProperty could be left as it is, because it's not nullable here. On other hand it would be consistent to have three-state behavior in this control as well.

ToggleButton is three-state because it is a primitive and used in tons of places. There are cases where you might want indeterminate 1. parent when children are both checked/unchecked 2. no selection is made. The ToggleSplitButton has much more narrow use cases and two states makes sense here IMO. It also matches the upstream API and what people are familiar with. I plan to leave this as-is unless real-world use cases appear where three state is needed.

Reply:
> It also matches the upstream API

Don't forget that even more avalonia developers are familiar with Avalonia API, and used to other common controls.

But yes, we can leave it as is. Not as big breaking change if we will have to change it.

Reply:
While Microsoft flavors of XAML have far more users, I guess my point was more about the fundamental design of the control. A ToggleSplitButton was designed for a very specific use case -- which is just as relevant to Avalonia. Basically, it's used to control whether a feature is on or off with a drop down of options (bullet list button). Uses outside of this probably should be a different control to begin with.